Fix cookie 'SameSite' attribute warning in Firefox (#599)

iipajcii opened the pull request:
This fix adds the SameSource Attribute and the Secure Attribute to the cookie test.
SameSource is set to 'Strict' so that only the website that the cookie will have access to it, making it a first-party cookie according to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite).

Finesse commented:
@iipajcii, thank you for your contribution. You've added both `SameSize` and `Secure` attributes, while Firefox requires either `Secure` or `SameSite`. Looks like your change makes the `cookieEnabled` component always return `false` on `http://` websites, and this can be avoided. Have you checked it?

iipajcii commented:
> @iipajcii, thank you for your contribution. You've added both `SameSize` and `Secure` attributes, while Firefox requires either `Secure` or `SameSite`. Looks like your change makes the `cookieEnabled` component always return `false` on `http://` websites, and this can be avoided. Have you checked it?

No I have not tried it on `https`, my development environment uses `http'. I could leave `SameSite` set to `Strict` and leave out `Secure`, however I'm not sure if the maintainers intend for this script to only be used in one site, or if it's okay for third party sites to use it.

If third-parties are to have access and if it needs to work over `https` I think setting `SameSite` to `Lax` would work

Finesse commented:
@iipajcii

> No I have not tried it on `https`, my development environment uses `http`.

I meant you to try HTTP usage, not HTTPS. You've set `Secure` flag, it means that the cookie will be available only on HTTPS pages and therefore the source will always return `false` on HTTP pages. Have you checked it? You've said that you've tried it on an HTTP page; have you tried to disable/enable cookies in browser settings and see if the source returned value changes?

> I could leave `SameSite` set to `Strict` and leave out `Secure`, however I'm not sure if the maintainers intend for this script to only be used in one site, or if it's okay for third party sites to use it. If third-parties are to have access and it needs to work over `https` I think setting `SameSite` to `Lax` would work

The FingerprintJS library is intended to be used on any website, both on HTTP and HTTPS. I think any `SameSite` value other than `None` will work because the cookie doesn't participate in HTTP requests because it's removed right after creation.

iipajcii commented:
> The FingerprintJS library is intended to be used on any website, both on HTTP and HTTPS. I think any `SameSite` value other than `None` will work because the cookie doesn't participate in HTTP requests because it's removed right after creation

I understand your point, since it's only a test to see if cookies work I've just updated my branch and removed `Secure` and now it only sets `SameSite` to `Strict`.

You should see the change in the 'Files Changed' Section now.

Finesse commented:
All good, but the changes must go to the `master` branch instead of the `next-sources` branch
